### DISCUSSION TITLE
Why is convenient asList() deprecated in favor of verbose asInstanceOf(InstanceOfAssertFactories.LIST)?

### DISCUSSION THREAD

**Discussion by NadChel** [assertj/assertj#3424] [Q&A] [answered]
Why is the convenient `asList()` deprecated in favor of the verbose `asInstanceOf(InstanceOfAssertFactories.LIST)` in `org.assertj.core.api.AbstractAssert`? All the former does is literally calling the latter while also allowing for cleaner client code

What are the reasons behind that decision?

AssertJ 3.25.3

**Comment by scordio** [comment, marked as answer]
It's mostly a design decision to draw a clear line for maintenance.

In short, `asList` is a historical method introduced before `asInstanceOf`/`InstanceOfAssertFactories` were available.

Given that `asList` is not type-safe (#3105) and many other related methods are not offered today (#1020), we wanted to draw the line knowing that we provide over 80 base factories plus more than 20 type-safe ones with `InstanceOfAssertFactories`.

About verbosity, a static import can of course help:

```java
asInstanceOf(LIST)
```

However, I agree `InstanceOfAssertFactory` tends to be verbose so we'll look at improving the naming in version 4 (#3425).
